components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -107,63 +172,145 @@ private static InternalAPI createAPI(String classFQName) {
+        if (isEnabled) {
+            internalInboundHttpPortProperty = properties.getProperty(Constants.INTERNAL_HTTP_API_PORT);
+            internalInboundHttpsPortProperty = properties.getProperty(Constants.INTERNAL_HTTPS_API_PORT);
         }
+        return isEnabled;
+    }
 
-        String internalInboundPortProperty =
-                properties.getProperty(Constants.INTERNAL_HTTP_API_PORT);
-        if (internalInboundPortProperty != null) {
-            try {
-                internalInboundPort = Integer.parseInt(internalInboundPortProperty);
-            } catch (Exception ex) {
-                handleException(Constants.INTERNAL_HTTP_API_PORT + " is not in proper format", ex);
+    private static SSLConfiguration setSslConfig(OMElement sslConfig) {
+
+        Iterator iterator = sslConfig.getChildElements();
+
+        String trustStore = null;
+        String keyStore = null;
+        String clientAuth = null;
+        String httpsProtocols = null;
+        String revocationVerifier = null;
+        String sslProtocol = null;
+        String prefferedCiphers = null;
+
+        while (iterator.hasNext()) {
+
+            OMElement parameter = (OMElement) iterator.next();
+
+            if (parameter.getFirstElement() != null) {
+                if (KEYSTORE_ATT.equals(parameter.getAttributeValue(NAME_ATT))) {
+                    keyStore = parameter.getFirstElement().toString();
+                } else if (TRUSTSTORE_ATT.equals(parameter.getAttributeValue(NAME_ATT))) {
+                    trustStore = parameter.getFirstElement().toString();
+
+                } else if (CERTIFICATE_REVOCATION_VERIFIER_ATT.equals(parameter.getAttributeValue(NAME_ATT))) {
+                    revocationVerifier = parameter.getFirstElement().toString();
+                } else {
+                    handleException("Invalid parameter found for internal API ssl configuration");
+                }
+            } else {
+                if (SSL_PROTOCOL_ATT.equals(parameter.getAttributeValue(NAME_ATT))) {
+                    sslProtocol = parameter.getText();
+                } else if (PREFERRED_CIPHERS_ATT.equals(parameter.getAttributeValue(NAME_ATT))) {
+                    prefferedCiphers = parameter.getText();
+                } else if (HTTPS_PROTOCOLS_ATT.equals(parameter.getAttributeValue(NAME_ATT))) {
+                    httpsProtocols = parameter.getText();
+                } else if (SSL_VERIFY_CLIENT_ATT.equals(parameter.getAttributeValue(NAME_ATT))) {
+                    clientAuth = parameter.getText();
+                } else {
+                    handleException("Invalid parameter found for internal API ssl configuration");
+                }
             }
         }
-        return internalInboundPort;
-    }
 
-    private static void handleException(String msg) {
-        log.error(msg);
-        throw new SynapseException(msg);
+        if (keyStore == null) {
```

Comment:
We can remove this line if we set the correct defaults

Reply:
We are not setring defaults